### PR TITLE
[8.0.1] Adjust indices.recovery.max_bytes_per_sec according to external settings

### PR DESCRIPTION
**Pull request by tlrx**
This pull request backports the node bandwidth settings merged in 8.1.0.

It cherry-picks the following changes:
- #82819
- #83350
- #83359 
- #83361
- #83372

### CODE REVIEW

**Review by henningandersen — APPROVED** [review]
LGTM

File: docs/changelog/82819.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 82819
+summary: "[Draft] Adjust `indices.recovery.max_bytes_per_sec` according to external\
```
Comment (henningandersen) [inline docs/changelog/82819.yaml:2]: Not sure if this auto-updates but if not, please remove `[Draft]`

File: docs/changelog/83350.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pr: 83350
```
Comment (henningandersen) [inline docs/changelog/83350.yaml:1]: I wonder if we should remove this changelog entry?

**Review by henningandersen — APPROVED** [review]
LGTM.

**Review by DaveCTurner — COMMENTED** [review]
LGTM

File: docs/changelog/83350.yaml (hunk 2 above)
Comment (tlrx) [reply docs/changelog/83350.yaml]: Done

**Pull request merged**